components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/dao/impl/RemoteFetchConfigurationDAOImpl.java

```diff
@@ -208,33 +214,135 @@ public void deleteRemoteFetchConfiguration(String configurationId, int tenantId)
+                                            resultSet.getInt(8));
+                                    Timestamp lastDeployed = resultSet.getTimestamp(9);
+                                    if (lastDeployed != null) {
+                                        obj.setLastDeployed(new Date(lastDeployed.getTime()));
+                                    }
+                                    return obj;
+                                })
+                                , preparedStatement -> {
+                                    preparedStatement.setInt(1, tenantId);
+                                    preparedStatement.setInt(2, offset + limit);
+                                    preparedStatement.setInt(3, offset);
+                                })
+                );
+
+            } else if (databaseProductName.contains(SQLConstants.DB_MSSQL)) {
+
+                return jdbcTemplate.withTransaction(template ->
+                        template.executeQuery(SQLConstants.LIST_BASIC_CONFIGS_BY_TENANT_MSSQL,
+                                ((resultSet, i) -> {
+                                    BasicRemoteFetchConfiguration obj = new BasicRemoteFetchConfiguration(
+                                            resultSet.getString(1),
+                                            resultSet.getString(2).equals("1"),
+                                            resultSet.getString(3),
+                                            resultSet.getString(4),
+                                            resultSet.getString(5),
+                                            resultSet.getString(6),
+                                            resultSet.getInt(7),
+                                            resultSet.getInt(8));
+                                    Timestamp lastDeployed = resultSet.getTimestamp(9);
+                                    if (lastDeployed != null) {
+                                        obj.setLastDeployed(new Date(lastDeployed.getTime()));
+                                    }
+                                    return obj;
+                                })
+                                , preparedStatement -> {
+                                    preparedStatement.setInt(1, tenantId);
+                                    preparedStatement.setInt(2, offset);
+                                    preparedStatement.setInt(3, limit);
+                                })
+                );
+
+            } else if (databaseProductName.contains(SQLConstants.DB_POSTGRESQL)
+                    || databaseProductName.contains(SQLConstants.DB_DB2)) {
+
+                return jdbcTemplate.withTransaction(template ->
+                        template.executeQuery(SQLConstants.LIST_BASIC_CONFIGS_BY_TENANT_POSTGRES_DB2,
+                                ((resultSet, i) -> {
+                                    BasicRemoteFetchConfiguration obj = new BasicRemoteFetchConfiguration(
```

Comment:
Can we move this and code in line 290 to single function. Looks they do the same